docs/default.nix

```diff
@@ -1,33 +1,38 @@
 {
-  pkgs,
-  astal,
+  flakePkgs,
+  stdenvNoCC,
+  formats,
+  gi-docgen,
+  glib,
+  json-glib,
+  gobject-introspection,
+  gtk3,
+  gtk-layer-shell,
+  gdk-pixbuf,
+  libdbusmenu-gtk3,
+  wireplumber,
+  networkmanager,
+  ...
 }: let
-  toTOML = (pkgs.formats.toml {}).generate;
+  toTOML = (formats.toml {}).generate;
 
-  gen = pkg: name: out: toml: ''
-    mkdir -p $out/${out}
-    gi-docgen generate -C ${toTOML name toml} ${astal.${pkg}.dev}/share/gir-1.0/${name}-0.1.gir
-    cp -r ${name}-0.1/* $out/${out}
-  '';
+  genRefForPkg = {
+    name,
+    pkg,
+    outPath,
+    metaData,
+  }: let
+    tomlMetaData = toTOML name metaData;
+    devOutput = flakePkgs.${pkg}.dev;
+  in ''
+    mkdir -p $out/${outPath}
 
-  lib = name: namespace: description: {
-    authors ? "Aylur",
-    dependencies ? {},
-    out ? "libastal/${name}",
-  }:
-    gen name "Astal${namespace}" out {
-      library = {
-        inherit description authors;
-        license = "LGPL-2.1";
-        browse_url = "https://github.com/Aylur/Astal";
-        repository_url = "https://github.com/Aylur/Aylur.git";
-        website_url = "https://aylur.github.io/astal";
-      };
-      dependencies = {
-        inherit (dependency) "GObject-2.0";
-        inherit dependencies;
-      };
-    };
+    gi-docgen generate \
+      -C ${tomlMetaData} ${devOutput}/share/gir-1.0/${name}-0.1.gir
```

Comment:
I assume the version should also be parameterized (maybe in the future).

Reply:
The gir api version is not going to change in the near future I suppose, but the astal lib api version might go to 1.0 at some point, but definitely not in the near future, its fine for now I think